ui/app/components/totp-edit.js

```diff
@@ -57,29 +64,45 @@ export default class TotpEdit extends Component {
 
   @action
   reset() {
-    this.model.unloadRecord();
-    this.successCallback(null);
+    this.args.model.unloadRecord();
+    this.transitionToRoute(SHOW_ROUTE, this.args.model.name);
   }
 
   @action
-  delete() {
-    this.persist('destroyRecord', () => {
+  async deleteKey() {
+    try {
+      await this.args.model.destroyRecord();
       this.transitionToRoute(LIST_ROOT_ROUTE);
-    });
-  }
-
-  @action
-  create(event) {
-    event.preventDefault();
-    const modelId = this.model.name;
-    // prevent from submitting if there's no key
-    // maybe do something fancier later
-    if (isBlank(modelId)) {
+    } catch (err) {
+      // err will display via model state
       return;
     }
-
-    this.persist('save', () => {
-      this.transitionToRoute(SHOW_ROUTE, modelId);
-    });
+    this.flashMessages.success('Successfully deleted key');
   }
+
+  createKey = task(
+    waitFor(async (event) => {
+      event.preventDefault();
+      const { isValid, state, invalidFormMessage } = this.args.model.validate();
+      this.modelValidations = isValid ? null : state;
+      this.invalidFormAlert = invalidFormMessage;
+
+      if (!isValid) return;
+      try {
+        const allFields = [...this.keyFormFields, ...Object.values(this.groups).flat()];
```

Comment:
usually we refer to fields outside toggle groups as "default" fields. I wonder if renaming the keyFormFields would make that distinction clearer. Not a big deal! so fine to leave as is
```suggestion
        const allFields = [...this.defaultKeyFormFields, ...Object.values(this.groups).flat()];
```